js/interface.js

```diff
@@ -1720,6 +1729,75 @@ $('#show-versions').click(function() {
   fetchCurrentDataSourceVersions();
 });
 
+function findSecurityRule() {
+  var rule = currentDataSourceRules.map(rule => {
+    if (rule.allow.hasOwnProperty('tokens') && rule.allow.tokens.indexOf(widgetData.tokenId) !== -1) {
```

Comment:
This makes it a lot simpler to read and write. See our coding standards on function writing.

```suggestion
    var tokens = _.get(rule, 'allow.tokens');
    
    if (!tokens || tokens.indexOf(widgetData.tokenId) === -1) {
      return;
    }
    
    return rule;
```

Reply:
Did the changes as per your suggestions.